src/js/angular/resources/chart-models/chart-data.js

```diff
@@ -0,0 +1,26 @@
+export class ChartData {
+    constructor(translateService, chartOptions) {
+        this.translateService = translateService;
+        this.range = 150;
+        this.chartOptions = chartOptions;
+        this.chartSetup();
+        this.data = this.createDataHolder();
+    }
+    chartSetup() {}
+    createDataHolder() {
+        throw new Error('Must implement data holder creation');
+    }
+    addData(timestamp, data) {
+        this.removeOldData();
+        this.addNewData(timestamp, data);
+        this.updateRange();
+    }
+
+    removeOldData() {
+        if (this.data[0].values.length > this.range) {
+            this.data.forEach((data) => data.values.shift());
```

Comment:
array.shift is expensive function would you consider something else?

Reply:
I am not sure which will be better in this case.

Reply:
Shift seems to be better in this case. I will leave it.